src/NativeScript/ObjC/Unmanaged/UnmanagedType.cpp

```diff
@@ -0,0 +1,51 @@
+#include "UnmanagedType.h"
+
+namespace NativeScript {
+using namespace JSC;
+
+const ClassInfo UnmanagedConstructor::s_info = { "Unmanaged", &Base::s_info, 0, CREATE_METHOD_TABLE(UnmanagedConstructor) };
+
+void UnmanagedConstructor::finishCreation(VM& vm, JSGlobalObject* globalObject, JSCell* returnType) {
+    Base::finishCreation(vm);
+
+    this->_returnType.set(vm, this, returnType);
+
+    this->_ffiTypeMethodTable.ffiType = &ffi_type_pointer;
+    this->_ffiTypeMethodTable.read = &read;
+    this->_ffiTypeMethodTable.write = &write;
+    this->_ffiTypeMethodTable.canConvert = &canConvert;
+    this->_ffiTypeMethodTable.encode = &encode;
+}
+
+JSValue UnmanagedConstructor::read(ExecState* execState, const void* buffer, JSCell* self) {
+    const void* data = *reinterpret_cast<void* const*>(buffer);
+
+    if (!data) {
+        return jsNull();
+    }
+
+    GlobalObject* globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+    UnmanagedConstructor* constructor = jsCast<UnmanagedConstructor*>(self);
+
+    return UnmanagedInstance::create(execState->vm(), UnmanagedInstance::createStructure(globalObject, globalObject->unmanagedInstanceStructure()->storedPrototype()), constructor->_returnType.get(), const_cast<void*>(data));
```

Comment:
I think that the idea of caching the structure in the global object is not to create one for each unmanaged instance, so `globalObject->unmanagedInstanceStructure()` should suffice for the second argument.